File: src/app/core/singleton-services/movies/movie.ts
import { MovieMetadata } from "./MovieMetadata";

export class Movie {
  public movieId!: string;
  public title!: string;
  public movieMetadata: MovieMetadata | undefined;

  constructor(movieId: string, title: string){
    this.movieId = movieId;
    this.title = title;
  }
}

File: fake/Dii_TheaterManagement_Bff.Fake/Data/Movie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace FakeTheaterBff.Data
{
    public class Movie
    {
        public Guid MovieId { get; set; }

        public virtual ICollection<Booking> Bookings { get; set; }

        public MovieMetadata MovieMetadata { get; set; }

        [Required]
        public string Title { get; set; }
    }
}
